Hi.

The loop in the i/o core that submits I/O lost the check for aborting in
the recent cleanup.

This patch restores it.

Nigel

 io.c |    3 +++
 1 files changed, 3 insertions(+)
diff -ruNp 9030-listen-to-abort-request.patch-old/kernel/power/io.c 9030-listen-to-abort-request.patch-new/kernel/power/io.c
--- 9030-listen-to-abort-request.patch-old/kernel/power/io.c	2005-12-27 20:44:56.000000000 +1000
+++ 9030-listen-to-abort-request.patch-new/kernel/power/io.c	2005-12-26 21:42:20.000000000 +1000
@@ -315,6 +315,9 @@ static int do_rw_loop(int write, int fin
 		/* Interactivity*/
 		check_shift_keys(0, NULL);
 
+		if (test_result_state(SUSPEND_ABORTED) && write)
+			return 1;
+
 		/* Prepare next */
 		current_page_index = get_next_bit_on(*pageflags,
 				current_page_index);



_______________________________________________
Suspend2-devel mailing list
Suspend2-devel@lists.suspend2.net
http://lists.suspend2.net/mailman/listinfo/suspend2-devel

